bitnami/kube-prometheus/values.yaml

```
@@ -829,6 +829,12 @@ prometheus:
   ## ref: https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config
   ##
   scrapeTimeout: ""
+  ## @param prometheus.sampleLimit Per-scrape max number of scraped samples. Requires Prometheus v2.45.0 and newer
+  ## ref: https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config
+  sampleLimit: 0
+  ## @param prometheus.enforcedSampleLimit Override sampleLimits set by ServiceMonitor, PodMonitor or Probe objects
+  ## ref: https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#monitoring.coreos.com/v1.PrometheusSpec
+  enforcedSampleLimit: 0
```

We shouldn't set a value by default here, as this will cause to always have `sampleLimit` and `enforcedSampleLimit` in the `prometheus.yaml` template. It's true the Prometheus' config default is `sample_limit=0` but we don't need it explicitly set by default in the chart.

```suggestion
  sampleLimit: ""
  ## @param prometheus.enforcedSampleLimit Override sampleLimits set by ServiceMonitor, PodMonitor or Probe objects
  ## ref: https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#monitoring.coreos.com/v1.PrometheusSpec
  enforcedSampleLimit: ""
```